src/util/strutil.h

```diff
@@ -7,19 +7,41 @@
 #pragma once
 
 // return utf8 encoded length of string
-int str_length(const char *str);
+int str_length
+(
+	const char *str
+);
 
-//convert str to a lower case string and save it in lower
-void str_tolower(const char *str, char *lower, size_t *lower_len);
+// return true if utf8 string is valid
+bool str_utf8_validate
+(
+    const char *str
+);
 
-//convert str to a n upper case string and save it in upper
-void str_toupper(const char *str, char *upper, size_t *upper_len);
+// convert ascii str to a lower case string and save it in lower
+void str_tolower_ascii
+(
+	const char *str,
+	char *lower,
+	size_t *lower_len
+);
 
+// return utf8 string converted to lower case
+char *str_tolower
+(
+	const char *str
+);
```

Comment:
I'll assume passing an ASCII string to this function will also work, and the returned string will remain ASCII?
if that's the case please update the comment accordingly as it is a bit confusing in such case.

Reply:
ASCII is a subset of UTF-8, not an alternative